src/controllers/problem.ts

```diff
@@ -54,11 +56,25 @@ const problemController = {
             res.status(statusCodes.MISSING_PARAMS).json(errors.formatWith(errorMessage).array()[0]);
         } else {
             try {
+                // TODO: add verification that the fields within the Object problemMetadata are present
                 const problemData: IProblem = {
                     ...req.body,
-                    problemId: calculateProblemHash(req.body.problemId, req.body.source)
+                    problemId: calculateProblemHash(req.body.source.toUpperCase(), req.body.problemMetadata.platformProblemId.toUpperCase())
```

Comment:
we don't want to uppercase every platformProblemId seeing how some might be case sensitive. for now, uppercase if platform is codeforces